### DISCUSSION TITLE
Adding a new device

### DISCUSSION THREAD

**Discussion by sanjuanp** [PlusToolkit/PlusLib#1063] [Q&A]
Hello I am doing a project on US with an EM tracker. I am following the instructions of the guide, I have written de Find for the Module folder and I am doing the vtkPlusYourDeviceNameSource.cxx; but I do not know how to test if everything I am doing is working, I do not know how can I open the application with my new device. I don't know how can I create a .exe, do I have to compile the PlusServerLauncher folder?

**Comment by Sunderlandkyl** [comment]
If you've compiled PlusLib, you can run a device using PlusServer from command line: http://perk-software.cs.queensu.ca/plus/doc/nightly/user/ApplicationPlusServer.html.

**Reply by sanjuanp** [reply]
I can not compile PlusLib, it seems there is an error in the line 225 of the CMakeLists.txt, the error is the following: 

Error CMake Error at C:\Users\...\source\PLUSTOOLKIT\CMakeLists.txt:225 (IF):if given arguments:```"MSVC" "AND" "MATCHES" "x64"``` Unknown arguments specified. 

I don't know if there is something that a should install. But I have tried with other different CMake projects and I do not have this problem. Maybe it is because of the version of my VS, could it be?

On the other hand, in this case I have cloned the PlusLib repository, but, once I get to compile PlusLib, do I have to clone the rest of the repositories in the same folder and open PlusServer? 

I am an student and I have never worked with CMake.

**Reply by Sunderlandkyl** [reply]
The easiest way to build Plus would be to follow the instructions in the PlusBuild repository: https://github.com/PlusToolkit/PlusBuild/blob/master/Docs/BuildInstructionsWindows.md.

**Reply by sanjuanp** [reply]
Thank you.
